vm/systemSmartContracts/delegation.go

```diff
@@ -2599,6 +2615,128 @@ func (d *delegation) getMetaData(args *vmcommon.ContractCallInput) vmcommon.Retu
 	return vmcommon.Ok
 }
 
+func (d *delegation) basicCheckForLiquidStaking(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
```

Comment:
missing tests

Reply:
next PRs. want to have the full picture first